R/epi_recipe.R

```diff
@@ -273,12 +287,65 @@ is_epi_recipe <- function(x) {
 #'   add_epi_recipe(r)
 #'
 #' workflow
+#'
+#' r2 <- epi_recipe(jhu) %>%
+#'   step_epi_lag(death_rate, lag = c(0, 7, 14)) %>%
+#'   step_epi_ahead(death_rate, ahead = 7)
+
+#' workflow <- update_epi_recipe(workflow, r2)
+#'
+#' workflow <- remove_epi_recipe(workflow)
+#'
+#' workflow
+#'
+#' # Additional feature in `update_epi_recipe` is to change a step
+#' # in the recipe from the workflow
+#' workflow <- epi_workflow() %>%
+#'   add_epi_recipe(r)
+#'
+#' # Update second step, `step_epi_ahead()`, to have an
+#' # ahead value of 14 days
+#' workflow = workflow %>% update_epi_recipe(step_num = 2, ahead = 14)
+#' workflows::extract_preprocessor(workflow)
+#'
 add_epi_recipe <- function(
     x, recipe, ..., blueprint = default_epi_recipe_blueprint()) {
   workflows::add_recipe(x, recipe, ..., blueprint = blueprint)
 }
 
+#' @rdname add_epi_recipe
+#' @export
+remove_epi_recipe <- function(x) {
+  workflows:::validate_is_workflow(x)
+
+  if (!workflows:::has_preprocessor_recipe(x)) {
+    rlang::warn("The workflow has no recipe preprocessor to remove.")
+  }
 
+  actions <- x$pre$actions
+  actions[["recipe"]] <- NULL
+
+  new_epi_workflow(
+    pre = workflows:::new_stage_pre(actions = actions),
+    fit = x$fit,
+    post = x$post,
+    trained = FALSE
+  )
+}
+
+#' @rdname add_epi_recipe
+#' @export
+update_epi_recipe <- function(
```

Comment:
I think we should do this a bit differently. Let's create:

1. `replace_epi_recipe()`. It removes the recipe from the workflow and then adds the new one.
2. `update_epi_recipe()` (maybe even `adjust_epi_recipe()`). This should be an S3 generic with methods `.workflow()` and `.recipe()`. Then args can be `x`, `step_num` (mandatory), `...`, and `blueprint`. Then it can do the update to either kind of object.

Reply:
Done, but instead of `replace_epi_recipe()` used the name `update_epi_recipe()` to maintain consistency with the functions in the recipes package (particularly their `update_recipe()` function from https://github.com/tidymodels/workflows/blob/HEAD/R/pre-action-recipe.R, which I think operates in the same way).
 So, the name used for 2. is `adjust_epi_recipe()` (& correspondingly `adjust_frosting()`).